src/main/scala/org/apache/spark/sql/delta/DeltaOptions.scala

```diff
@@ -77,6 +77,15 @@ trait DeltaWriteOptionsImpl extends DeltaOptionParser {
   def canOverwriteSchema: Boolean = {
     options.get(OVERWRITE_SCHEMA_OPTION).map(toBoolean(_, OVERWRITE_SCHEMA_OPTION)).getOrElse(false)
   }
+
+  /**
+   * Whether to write new data to the table
+   * or just rearrange data that is already part of the table
+   */
+  def canChangeData: Boolean = {
```

Comment:
I'm not sure this is the right name for this method. Specifying this option doesn't really control whether new data is written to the table; it just declares that the data being written by that commit does not change data in the table. I'd call it "noDataChange" or "changesData" here.

Reply:
I will rename this method to noDataChange

Reply:
@mukulmurthy  
what do you think about adding "This option declares that the data being written by that commit does not change data in the table" in the java doc ?